gravitee-apim-console-webui/src/management/api/proxy/endpoints/components/endpoint-http-config/endpoint-http-config.component.ts

```diff
@@ -225,20 +243,31 @@ export class EndpointHttpConfigComponent implements OnInit, OnDestroy {
       )
       .subscribe((enabled) => {
         if (enabled === true) {
-          httpProxy.get('useSystemProxy').enable();
-          httpProxy.get('host').enable();
-          httpProxy.get('port').enable();
-          httpProxy.get('type').enable();
-          httpProxy.get('username').enable();
-          httpProxy.get('password').enable();
+          httpProxy.get('useSystemProxy').enable({ onlySelf: true });
+          httpProxy.get('host').addValidators(Validators.required);
+          httpProxy.get('host').enable({ onlySelf: true });
+          httpProxy.get('port').addValidators(Validators.required);
+          httpProxy.get('port').enable({ onlySelf: true });
+          httpProxy.get('type').enable({ emitEvent: false });
+          httpProxy.get('username').enable({ emitEvent: false });
+          httpProxy.get('password').enable({ emitEvent: false });
         } else {
-          httpProxy.get('useSystemProxy').disable();
-          httpProxy.get('host').disable();
-          httpProxy.get('port').disable();
-          httpProxy.get('type').disable();
-          httpProxy.get('username').disable();
-          httpProxy.get('password').disable();
+          httpProxy.get('useSystemProxy').disable({ emitEvent: false });
+          httpProxy.get('useSystemProxy').setValue(false, { onlySelf: true });
+          httpProxy.get('host').clearValidators();
+          httpProxy.get('host').disable({ onlySelf: true });
+          httpProxy.get('port').clearValidators();
+          httpProxy.get('port').disable({ onlySelf: true });
+          httpProxy.get('type').disable({ emitEvent: false });
+          httpProxy.get('username').disable({ emitEvent: false });
+          httpProxy.get('password').disable({ emitEvent: false });
         }
+
+        // Update validators
+        Object.keys(httpProxy.controls).forEach((controlName) => {
```

Comment:
I would have guess that `httpProxy.updateValueAndValidity({ emitEvent: false })` would have do the same than looping on form keys no ?
And I don't really get why after that you still need to call the updateValueAndValidity but send an event this time ?